doc/15-troubleshooting.md

```diff
@@ -230,6 +232,97 @@ icinga2 7764 nagios  mem    REG                8,5   744776 2631636 /usr/lib/x86
 #
 ```
 
+### Optimise apply rules and group assign conditions
+
+#### Remove actually unused apply rules
+
+If `icinga2 daemon -C` warns you like shown below and "it's okay" for you
+as everything behaves as desired, consider removing
+or commenting out the respective apply rule.
+
+```
+[...] warning/ApplyRule: Apply rule '...' (...) for type '...' does not match anywhere!
+```
+
+At least consider changing the `assign where` filter from, say
+`assign where my_complex_filter()`, to `assign where false && my_complex_filter()`.
+`&&` will evaluate `false` first and in an instant.
+If it's not true, `&&` will "return" `false` and skip `my_complex_filter()`.
```

Comment:
Why would you ever want to do that instead of just removing the rule?

Reply:
For keeping it "for later". To disable it temp-ly. IDK.

Reply:
Then you just comment it out? Or is it a Director thing that you can only delete but not disable?

Anyways, this section would be more useful if it covered more cases, like that `host.vars.production && match(...)` is probably faster than `match(...) & host.vars.production`.